src/traces/indicator/attributes.js

```diff
@@ -0,0 +1,408 @@
+            valType: 'color',
+            role: 'info',
+            dflt: colorAttrs.defaultLine,
+            editType: 'plot',
+            description: [
+                'Sets the color of the line enclosing each sector.'
+            ].join(' ')
+        },
+        width: {
+            valType: 'number',
+            role: 'info',
+            min: 0,
+            dflt: 0,
+            editType: 'plot',
+            description: [
+                'Sets the width (in px) of the line enclosing each sector.'
+            ].join(' ')
+        },
+        editType: 'calc'
+    },
+    thickness: {
+        valType: 'number',
+        role: 'info',
+        min: 0,
+        max: 1,
+        dflt: 1,
+        editType: 'plot',
+        description: [
+            'Sets the thickness of the bar as a fraction of the total thickness of the gauge.'
+        ].join(' ')
+    },
+    editType: 'calc'
+};
+
+var stepsAttrs = templatedArray('threshold', extendDeep({}, gaugeBarAttrs, {
+    range: {
+        valType: 'info_array',
+        role: 'info',
+        items: [
+            {valType: 'number', editType: 'plot'},
+            {valType: 'number', editType: 'plot'}
+        ],
+        editType: 'plot',
+        // impliedEdits: {'autorange': false},
+        description: [
+            'Sets the range of this axis.',
+            'If the axis `type` is *log*, then you must take the log of your',
+            'desired range (e.g. to set the range from 1 to 100,',
+            'set the range from 0 to 2).',
+            'If the axis `type` is *date*, it should be date strings,',
+            'like date data, though Date objects and unix milliseconds',
+            'will be accepted and converted to strings.',
+            'If the axis `type` is *category*, it should be numbers,',
+            'using the scale where each category is assigned a serial',
+            'number from zero in the order it appears.'
+        ].join(' ')
+    }
+}));
+
+module.exports = {
+    mode: {
+        valType: 'flaglist',
+        editType: 'calc',
+        role: 'info',
+        flags: ['number', 'delta', 'gauge'],
+        dflt: 'number'
+    },
+    value: {
+        valType: 'number',
+        editType: 'calc',
+        role: 'info',
+        anim: true,
+        description: [
+            'Sets the number to be displayed.'
+        ].join(' ')
+    },
+    align: {
+        valType: 'enumerated',
+        values: ['left', 'center', 'right'],
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the horizontal alignment of the `text` within the box.',
+            'Note that this attribute has no effect if an angular gauge is displayed:',
+            'in this case, it is always centered'
+        ].join(' ')
+    },
+    // position
+    domain: domainAttrs({name: 'indicator', trace: true, editType: 'calc'}),
+
+    vmin: {
+        valType: 'number',
+        editType: 'calc',
+        role: 'info',
+        dflt: 0,
```

Comment:
Interesting `dflt` here. I guess this make sense for most cases (e.g. for money), but maybe setting a min value could be called _too magical_.

Reply:
By the way, I also set `vmax` to 1.5 times the current value. This is going to break for negative values. What would be good defaults for `v(min|max)`?

Reply:
Sorry for the confusion, I was under the impression that `vmin` and `vmax` were also used to constraint the `mode: 'number'` text rendering. 

Your `vmin: 0` and `vmax: 1.5 * value` default sounds ok for `mode: 'gauge'`, but then again maybe `vmin` and `vmax` should be under `gauge.vmin` and `gauge.vmax` :thinking:

Reply:
> Sorry for the confusion, I was under the impression that `vmin` and `vmax` were also used to constraint the `mode: 'number'` text rendering.

It used to be the case but no more: we now resize the text by measuring its actual size in the viewport.

> Your `vmin: 0` and `vmax: 1.5 * value` default sounds ok for `mode: 'gauge'`, but then again maybe `vmin` and `vmax` should be under `gauge.vmin` and `gauge.vmax`

Yes, let's move `v(min|max)` under `gauge` where they belong.

Reply:
> Yes, let's move `v(min|max)` under `gauge` where they belong.

:+1:

Reply:
Or maybe `gauge.range` would be better and more consistent with `gauge.steps[i].range` :thinking:

Reply:
I chose to replace attributes `v(min|max)` with a single attribute `gauge.axis.range`.

See commit 95de45724b62a7a1c3d22434af838bd8c690cbfe

Reply:
:ok_hand: